lib/sycamore/sycamore/transforms/base.py

```diff
@@ -49,6 +49,21 @@ def __init__(
         enable_auto_metadata: bool = False,
         **resource_args,
     ):
+        if child is None:
+            logging.info("Assuming this is for local execution only, not checking serializability")
+        else:
+            # If serializability fails, the error messages are very confusing. These checks
+            # give a much more sensible error message and give it before ray starts execution.
+            from ray.util import inspect_serializability
+            import io
+
+            log = io.StringIO()
+            (ok, s) = inspect_serializability(
+                [f, name, args, kwargs, constructor_args, constructor_kwargs], print_file=log
+            )
+            if not ok:
+                raise ValueError(f"Something isn't serializable {s}\nLog: {log.getvalue()}")
```

Comment:
Any chance we can also stick the name of the failing node/map fn in this error message?